File: code/backend/php/src/Core/Domain/Story/TraitInfo.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class TraitInfo
{
    public function __construct(
        public string $uuid,
        public ?string $name = null,
        public ?string $description = null,
        public int $costPositive = 0,
        public int $costNegative = 0,
        public ?int $idClassPermitted = null,
        public ?int $idClassProhibited = null,
        public ?int $idCard = null,
        public ?CardInfo $card = null,
        public int $life = 0,
        public int $energy = 0,
        public int $sad = 0,
        public int $dexterity = 0,
        public int $intelligence = 0,
        public int $constitution = 0,
        public int $weight = 0
    ) {
    }
}

File: code/backend/python/app/core/models/story/trait_info.py
from dataclasses import dataclass
from app.core.models.story.card_info import CardInfo

@dataclass
class TraitInfo:
    uuid: str
    name: str | None = None
    description: str | None = None
    costPositive: int = 0
    costNegative: int = 0
    idClassPermitted: int | None = None
    idClassProhibited: int | None = None
    idCard: int | None = None
    card: CardInfo | None = None
    # v0.35.2 — true when the trait is not offered at character creation. Reported, never
    # filtered: the same list resolves the traits a character already has.
    hideOnStartMatch: bool = False
    life: int = 0
    energy: int = 0
    sad: int = 0
    dexterity: int = 0
    intelligence: int = 0
    constitution: int = 0
    weight: int = 0
